packages/search-input/src/SearchInput/SearchInput.styles.ts

```diff
@@ -35,26 +34,29 @@ export const inputWrapperStyle = css`
 `;
 
 export const inputWrapperSizeStyle: Record<Size, string> = {
+  [Size.XSmall]: css`
+    font-size: ${typeScales.body1.fontSize}px;
+    line-height: ${typeScales.body1.lineHeight}px;
+    height: 22px;
+    grid-template-columns: 28px 1fr 36px;
+  `,
   [Size.Small]: css`
     font-size: ${typeScales.body1.fontSize}px;
     line-height: ${typeScales.body1.lineHeight}px;
     height: 28px;
-    grid-template-columns: 28px 1fr;
-    grid-auto-columns: 28px;
+    grid-template-columns: 32px 1fr 36px;
   `,
   [Size.Default]: css`
     font-size: ${typeScales.body1.fontSize}px;
     line-height: ${typeScales.body1.lineHeight}px;
     height: 36px;
-    grid-template-columns: 36px 1fr;
-    grid-auto-columns: 36px;
+    grid-template-columns: 36px 1fr 36px;
   `,
   [Size.Large]: css`
-    font-size: 18px;
-    line-height: 32px;
+    font-size: ${typeScales.large.fontSize}px;
+    line-height: ${typeScales.large.lineHeight}px;
     height: 48px;
-    grid-template-columns: 48px 1fr;
-    grid-auto-columns: 48px;
+    grid-template-columns: 36px 1fr 44px;
```

Comment:
dropped `grid-auto-columns` in favor of setting a predefined space for the clear button because I observed unwanted layout shifts at the large size

https://github.com/mongodb/leafygreen-ui/assets/52300529/47dd188d-a775-4f6d-8a29-0d2f55a0f7ab